quickwit-doc-mapping/src/default_doc_mapper/default_mapper.rs

```diff
@@ -0,0 +1,377 @@
+/*
+    Quickwit
+    Copyright (C) 2021 Quickwit Inc.
+
+    Quickwit is offered under the AGPL v3.0 and as commercial software.
+    For commercial licensing, contact us at hello@quickwit.io.
+
+    AGPL:
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use super::{
+    field_mapping_entry::DocParsingError, resolve_field_name, FieldMappingEntry, FieldMappingType,
+};
+use crate::DocMapper;
+use anyhow::bail;
+use quickwit_proto::SearchRequest;
+use serde::{Deserialize, Serialize};
+use serde_json::{self, Value as JsonValue};
+use std::{
+    collections::{HashMap, HashSet},
+    convert::TryFrom,
+    fs::File,
+    io::BufReader,
+    path::Path,
+};
+use tantivy::{
+    query::Query,
+    schema::{Field, FieldEntry, FieldType, FieldValue, Schema, SchemaBuilder, Type, STORED},
+    Document,
+};
+
+static SOURCE_FIELD_NAME: &str = "_source";
+
+/// DefaultDocMapperBuilder is here
+/// to create a valid DefaultDocMapper.
+#[derive(Default, Serialize, Deserialize, Clone)]
+pub struct DefaultDocMapperBuilder {
```

Comment:
I'm confused... The builder is only used for deserialization or is it meant for users as well?

If it is meant for users, I don't see hwo to set default_search_fields etc.

Reply:
Ah yes this is missing indeed, at first I coded it for deserialization and I thought it would be cool to use it to build your own mapper programmaticaly.

Reply:
I added a todo to complete or remove.